### DISCUSSION TITLE
Upgrading RS8 firmware

### DISCUSSION THREAD

**Discussion by txjmb** [aqualinkd/AqualinkD#311] [Q&A] [answered]
I have an old RS8 pool and spa that is Rev F.  I have been using aqualinkd for a long time, but would like to control a Pentair Intelliflo VS.   I know I have to upgrade the firmware to do that.  Would I just have to upgrade the CPU PCB?  The upgrade kits often include the power center board and a bezel, but everything is working well, so can I get away with simply swapping that one board with one I can get on eBay?

**Comment by sfeakes** [comment, marked as answer]
For VSP you need Rev Q or newer.

Rev `F` board uses a 44 pin socketed chip that can be replaced, Replacing the chip will only get you to Rev `I` though.
From Rev `J` on they moved to a 52pin socket. (that can be replaced to get to Rev `M`)
From Rev `N` they moved to a soldered CPU that can't be replaced. (but can be upgraded with a daughter board)

If you want VSP, look for a board that’s rev Q or newer, rev O was when they first introduced VSP support, but it was crap and didn't support many VSP. (I'm sure a Pentair VSP would have issues).

So any of the cheaper (chip  or daughter board) upgrades are not going to work for you.   You will want one that includes the power center board and bezel.

**Reply by txjmb** [reply]
Thank you for the great information, I appreciate it very much.  Sounds like I'll be doing the full upgrade.  I have really enjoyed using Aqualinkd.

**Reply by sfeakes** [reply]
I forgot to add, if you are upgrading rev T (1 or 2) is about the best bang for the buck.  It’s relatively cheap since it’s a number of years old, and no decent features after that (that I know of)
